Update known limitations to include information about Executor support (#569)
astronomer/docs: astro/known-limitations.md

As we build and grow Astro, our team is committed to maintaining transparency about the current limitations of the product as well as the roadmap ahead of us.

The list below represents known issues at the time of writing. We're moving fast to address these and are excited to work with you to identify solutions to some of these foundational questions.

If you have questions or feedback about any item below, reach out to your Customer Success Manager.

## Known Limitations

> - Assistance from our team is required to give the first user in your Organization access to Astro.

Reviewer: ```suggestion
```

Deleting this, per Piotr - 

> No. Once the Organization is created, the user can log into the Organization and is granted a role.

- In order to invite a user to a Workspace, the user must already be a member of the Organization as described in [Add a User](add-user.md). Workspace invites that can add a user to the Organization are not currently supported.

> - When a user first creates an account, they will be asked to validate their email address. Email validation is not currently required to access Astro, but we encourage users to follow the process anyway as we will enforce it in the future.

Reviewer: ```suggestion
```

- If a user changes Workspace roles on Astro, it can take a maximum of 10 minutes for corresponding Airflow permission changes to take effect.

> - Astro supports only the Celery Executor. If you currently use the Kubernetes Executor, you may have to modify your tasks to run on Astro. A new, feature-rich executor built by Astronomer is coming soon.

Author: "feature-rich" is true, but should we frame this around Kube because that's the source of the limitation? Maybe we could say "task level resource configuration is coming soon"

Reviewer: I feeeeel like the point is `you may have to modify your tasks`, and there's a bullet point under "coming soon" below about it?

Author: Then in that case, I am gonna delete the "coming soon" here since it's repetitive of below.

- Clicking on **Refresh DAG** in the Airflow UI will redirect you to `<org-name>.astronomer.run` (Astro Home Page) instead of the task instance URL. We recommend upgrading to [Runtime 4.0](runtime-release-notes.md#astro-runtime-400), as Airflow 2.2 no longer requires this refresh button in the Airflow UI.

> - Deployment API keys cannot currently be used to automate updating the Deployment via the Astro CLI.

Author: This is not true no?As long as the env vars are accessible in the environment of the CLI, the deploy will be successful.  see https://docs.astronomer.io/astro/ci-cd#github-actions

Reviewer: Deploys yes, but they can't use it to do `astro deployment update` for example and programmatically change resources and what not

> - Deployment API keys cannot currently be used to automate updating the Deployment via the Astro CLI.

Reviewer: ```suggestion
- Deployment API keys can be used to automate deploys to Astro, but they cannot currently be used to programmatically update other properties of the Deployment via the Astro CLI.
```
